### PR TITLE
Nvidia plugin support

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: release_files/Dockerfile.ubuntu [hunk 1]
```diff
@@ -151,6 +153,12 @@ RUN apt-get update && \
     rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
 # for NCS
 RUN if [ -f /ovms/lib/hddl/etc/rules.d/97-myriad-usbboot.rules ]; then mkdir -p /etc/udev/rules.d/ && cp /ovms/lib/hddl/etc/rules.d/97-myriad-usbboot.rules /etc/udev/rules.d/ && ldconfig ; fi
+# for NVIDIA
```
Comment (dtrawins) [inline release_files/Dockerfile.ubuntu:156]: this should be conditional

File: docs/docker_container.md [hunk 2]
```diff
@@ -148,3 +148,16 @@ cd model_server
 make docker_build INSTALL_DRIVER_VERSION=21.38.21026
 ```
 If not provided, version 21.38.21026 is used for Redhat and 21.48.21782 is used for Ubuntu.
+
+## Using NVIDIA Plugin
```
Comment (dtrawins) [inline docs/docker_container.md:152]: that should be added to accelerators.md. Building option could stay here.

File: Dockerfile.redhat [hunk 3]
```diff
@@ -28,6 +28,7 @@ ARG DL_INSTALL_DIR=/opt/intel/openvino/deployment_tools
 ARG DL_DIR=/tmp
 ARG JOBS
 ARG CHECK_COVERAGE=1
+ARG NVIDIA=0
```
Comment (dtrawins) [inline Dockerfile.redhat:31]: is it tested on RH? I'm not aware of the cuda image based on ubi

File: Makefile [hunk 4]
```diff
@@ -71,13 +72,16 @@ DIST_OS_TAG ?= $(BASE_OS_TAG)
 
 ifeq ($(BASE_OS),ubuntu)
   BASE_OS_TAG=$(BASE_OS_TAG_UBUNTU)
-  BASE_IMAGE ?= ubuntu:$(BASE_OS_TAG_UBUNTU)
+  ifeq ($(NVIDIA),1)
```
Comment (atobiszei) [inline Makefile:75]: If NVIDIA==1 & use_binary_package ==1 throw error?

File: docs/accelerators.md [hunk 5]
```diff
@@ -214,3 +214,16 @@ THROUGHPUT
 ```
 
 > **NOTE**: currently, AUTO plugin cannot be used with `--shape auto` parameter while GPU device is enabled.
+
+## Using NVIDIA Plugin
+
+*Note:* To build container with NVIDIA plugin use command: 
+```bash
+   OV_SOURCE_BRANCH=2022.1.0 NVIDIA=1 OV_USE_BINARY=0 make docker_build
```
Comment (mzegla) [inline docs/accelerators.md:222]: In other places we start with `make` and argument come next. Can we keep that convention here as well?

File: DockerfileMakePackage [hunk 6]
```diff
@@ -57,11 +60,11 @@ libjavajpeg.so libjava.so libjawt.so libjdwp.so libjimage.so libjli.so libjsig.s
 liblcms.so libmanagement_agent.so libmanagement_ext.so libmanagement.so libmlib_image.so libnet.so libnio.so \
 libprefs.so librmi.so libsaproc.so libsctp.so libsplashscreen.so libsunec.so libsystemconf.so libunpack.so libverify.so libzip.so
 
-RUN if [ "$ov_use_binary" == "1" ] ; then true ; else exit 0 ; fi ; find /opt/intel/openvino/runtime/lib/intel64/ -iname '*.so*' -exec cp -vP {} /ovms_release/lib/ \;
-RUN if [ "$ov_use_binary" == "1" ] ; then true ; else exit 0 ; fi ; find /opt/intel/openvino/runtime/lib/intel64/ -iname '*.mvcmd*' -exec cp -v {} /ovms_release/lib/ \;
-RUN if [ "$ov_use_binary" == "1" ] ; then true ; else exit 0 ; fi ; find /opt/intel/openvino/runtime/3rdparty/ -iname '*.so*' -exec cp -vP {} /ovms_release/lib/ \;
-RUN if [ "$ov_use_binary" == "1" ] ; then true ; else exit 0 ; fi ; find /opt/opencv/lib/ -iname '*.so*' -exec cp -vP {} /ovms_release/lib/ \;
-RUN if [ "$ov_use_binary" == "1" ] ; then true ; else exit 0 ; fi ; cp /opt/opencv/share/licenses/opencv4/* /ovms/release_files/thirdparty-licenses/
```
Comment (mzegla) [inline DockerfileMakePackage:64]: Why is condition removed? Is it unified now?

File: Dockerfile.ubuntu [hunk 7]
```diff
@@ -189,6 +190,63 @@ RUN if [ "$ov_use_binary" = "1" ] && [ "$DLDT_PACKAGE_URL" = "" ] ; then true ;
...
+    set -exuo pipefail ; \
+    rm -f /etc/apt/apt.conf.d/docker-clean ; \
+    echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache ; \
+    apt clean ; \
+    apt update ; \
+    apt install -y \
+		gnupg2 \
+		software-properties-common; \
+    if [[ ${enable_tensorrt-} == "1" ]] ; then dpkg -i /nv-tensorrt-repo-*.deb ; fi; \
+	apt-key adv --keyserver-options http-proxy=http://proxy-us.intel.com:911 --fetch-keys  https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/3bf863cc.pub ; \
```
Comment (mzegla) [inline Dockerfile.ubuntu:214]: I don't think we should specify Intel proxy here

File: DockerfileMakePackage [hunk 8]
```diff
@@ -35,6 +36,8 @@ RUN mkdir -vp /ovms_release/lib/custom_nodes
 
 RUN if [ -d /ovms/src/custom_nodes/lib/${BASE_OS} ] ; then true ; else exit 0 ; fi ; cp /ovms/src/custom_nodes/lib/${BASE_OS}/*.so /ovms_release/lib/custom_nodes/
 RUN cp /ovms/metadata.json /ovms_release/
+
+RUN if [ "$NVIDIA" == "1" ]; then true ; else exit 0 ; fi ; cp -v /openvino/bin/intel64/Release/lib/libopenvino_nvidia_gpu_plugin.so /ovms_release/lib/
```
Comment (jasiu86) [inline DockerfileMakePackage:40]: Is it correct?

**Comment by dtrawins** [comment]
obsolete PR, feature is already included in `develop` branch

**Pull request closed without merging**